Update typography

msecret opened the pull request:
Had to make some tweaks to make the typography acceptable in current
design.

![screenshot_2016-10-07_11-41-26](https://cloud.githubusercontent.com/assets/1701077/19203904/ba3f3cf8-8c8e-11e6-90d3-008e26dbeec9.png)
![screenshot_2016-10-07_11-41-55](https://cloud.githubusercontent.com/assets/1701077/19203905/ba4005ca-8c8e-11e6-886e-bdcc09a51708.png)
![screenshot_2016-10-07_11-42-22](https://cloud.githubusercontent.com/assets/1701077/19203906/ba40f96c-8c8e-11e6-8aa4-65ba7f5634c8.png)
![screenshot_2016-10-07_13-01-34](https://cloud.githubusercontent.com/assets/1701077/19203907/ba454f08-8c8e-11e6-8a8d-fee2347defb4.png)

thisisdano committed:
added missing fonts...?

Commit:
Update typography

Had to make some tweaks to make the typography acceptable in current
design.

thisisdano committed:
changed formatting of stack slightly

Commit:
Fix linting errors.

thisisdano committed:
added mac system font to the cascade

adborden reviewed (approved):
Good from my perspective, up to @thisisdano to approve  :)

Commit:
Fix corrupted font files

The files weren't working, probably based on where I downloaded
them from github. Made sure to download raw files over again and
things are working again.

thisisdano commented:
Still workin' on fixing the issues with this change — especially in the docs. Hope to finish tomorrow.

thisisdano committed:
style changes to adapt to docs pages; dashboard fixes tk

thisisdano committed:
starting on tightening dashboard styles

thisisdano committed:
fixed up search button

thisisdano committed:
working pass on search; missing close box

thisisdano committed:
tweaks for tab nav and page-lede items

thisisdano committed:
rough implementation of page-dek element

thisisdano committed:
first linting pass

thisisdano committed:
fixed post-lint position error

thisisdano committed:
fixed close box

thisisdano committed:
fixed linting error

thisisdano commented:
Worked on docs and dashboard; now on to the landing page(s)

msecret commented on src/css/base/typography.scss:
```diff
@@ -4,7 +4,23 @@
+}
+
+a {
+  text-decoration: none;
```
So would this mean that the only indication something is a link is a difference in color? I believe that might be an accessibility problem.

thisisdano replied:
no, they have an underline, but it's a border-bottom — at least in the docs, gotta do more scoping/spec'ing to make it work properly in the dashboard

msecret commented on src/css/override_vars.scss:
```diff
@@ -23,29 +23,89 @@ $color-secondary-lightest:   #deeeef; // DEEF!
+$base-type-rem: #{$base * 2}rem;
+$base-type-px: #{$base * 2}px;
+
+$bx5: #{($base / 2 / $base-type)}rem;    // 4px
```
I'm not sure I like the name of these, mainly because I don't know what they mean. Can you rename to something more understandable?

thisisdano replied:
They mean increments of a base unit (in this case 8px) — `$bx5` is `base * 0.5` and `$b5x` is `base * 5` — it's a way to specify a common whitespace scale without doing rem calculations all day.

msecret replied:
What if we called them `basex5` etc?

thisisdano replied:
I'm happy to consider a different variable for whitespace scale, though.

Also, FYI, all the objects in my design comps will be spaced on multiples of 8px.

thisisdano replied:
thinking about this more.

maybe `$gx` is a good prefix ("grid times") and `05` is the way to go for half scale (as in Buzzfeed's Solid)

so: 

`$gx05`, `$gx1`, `$gx2` etc...

thisisdano replied:
yeah, I like that more

msecret commented on src/css/override_vars.scss:
```diff
@@ -23,29 +23,89 @@ $color-secondary-lightest:   #deeeef; // DEEF!
+$b9x: #{($base * 9 / $base-type)}rem;   // 72px
+$b10x: #{($base * 10 / $base-type)}rem; // 80px
+
+$sans-s1: (8rem / $base-type);    // 8px;
```
So for these variables, you're starting low and moving up in values which is the opposite of how HTML works. As in h1 is largest while here, s1 is largest. Is there a reason for that?

thisisdano replied:
just what I like — closer to something like basscss

thisisdano replied:
Actually, it may still be the opposite of basscss (whose shortcode I don't really like for type scale...) IDK, maybe it should go the other way... but I like to think bigger number equals bigger face. I'm just one man, tho!

thisisdano replied:
Still irritates me in markdown that smaller heds get more `#`s and thus more visual weight

thisisdano replied:
To elaborate further — I think of these variables as elements of size or scale, like shoe sizes or dress sizes (higher number is greater scale), rather than elements of hierarchy or rank, like headers or positions in a race (higher number is lesser rank).

They also directly correspond (same names, same sizes) to the styles and type scale objects in the Sketch file, which could make things (potentially) easier for development...

msecret commented on src/css/base/typography.scss:
```diff
@@ -13,88 +29,151 @@ h1, h2, h3, h4, h5, h6 {
+  color: $color-mediumgray;
+  font-family: $font-sans;
+  font-size: $sans-s5;
+  font-weight: 200;
```
I think this should be 400, as that's the lowest weight we have right now for the 18Franklin font.

thisisdano replied:
I think we also have 18Franklin Light, which is `200`

msecret replied:
Actually I just checked, 18Franklin Light is `300`, https://github.com/18F/cg-style/pull/164/files#diff-e56fe3386704ccaf26352285f872ddb6R8

thisisdano replied:
ah, cool

msecret commented on src/css/components/actions.scss:
```diff
@@ -6,12 +6,11 @@
-  &:visited {
-    color: $color-primary;
-  }
+  &:visited { color: $color-primary; }
```
Please keep the formatting to what it previously was as that's whats compliant with the CSS styleguide.

thisisdano replied:
what does this comment mean? about one-line rules? seemed like I was getting lint errors when single-rulers weren't a single line — but I really don't care. i'll do em however.

msecret replied:
Yes, they shouldn't be one-line rules. Let me know what linting errors you're getting.

msecret reviewed (commented):
All the images you've changed out have solid white backgrounds. Is there a reason for that?

thisisdano commented:
re: bgs -- I was having trouble getting them to output properly, so I added BGs. It wasn't going to affect the layout, so I left em in.

thisisdano committed:
changed base vars to grid

thisisdano committed:
updated font-weight: 200 to 300 for light weight

thisisdano committed:
further tweaks to vars

thisisdano committed:
fixed icons on landing

thisisdano committed:
changes for landing page

thisisdano committed:
fixed lint errors

thisisdano commented:
@msecret I think I'm done with this pass of style tweaks. I've also opened the PRs referenced above to apply once these new styles take effect.

There's still a LOT to be done on the dashboard to fix stuff that looks strange, but I think that's a job for another branch

thisisdano committed:
removed big logo variant, made default bigger

thisisdano committed:
removed big logo variant, made default bigger

thisisdano commented:
**landing**

![screen shot 2016-10-14 at 9 00 50 am](https://cloud.githubusercontent.com/assets/11464021/19395196/45fd2658-91f1-11e6-93de-c8489fe89121.png)

- - -

**docs**

![screen shot 2016-10-14 at 9 32 58 am](https://cloud.githubusercontent.com/assets/11464021/19395204/50ecda9a-91f1-11e6-832a-22cbfc953d4f.png)

- - -

**dashboard**

![screen shot 2016-10-14 at 9 41 48 am](https://cloud.githubusercontent.com/assets/11464021/19395445/75f23c76-91f2-11e6-8792-6c8e9e5d40b1.png)

msecret commented on npm-debug.log:
```diff
@@ -0,0 +1,43 @@
+0 info it worked if it ends with ok
```
@thisisdano this file needs to be removed

msecret commented on src/css/override_vars.scss:
```diff
@@ -14,34 +15,101 @@ $color-primary-alt-darkest:  darken($color-primary-alt, 70%);
+$grid-type: $grid * 2;
+$grid-type-px: #{$grid-type}px;
+
+$gx05: #{($grid / 2 / $grid-type)}rem;    // 4px
```
What's `gx`? Can we use full words, these variable names are too confusing.

thisisdano replied:
I outlined my reasoning above. It is `grid * multiplier` — what level of verbose-ness do you desire? `$gridx05` or `$grid-05` — feels bulky and unnecessary to me, esp for something entered very often, but at the end of the day I don't care that much.

msecret replied:
I would say either of those options work. I think the bulk is necessary here to ensure other developers that work on this code can easily understand these variables that are used throughout the library.

thisisdano replied:
ok, changing these to `$grid-05`, `$grid-1`...

msecret commented on src/css/override_vars.scss:
```diff
@@ -14,34 +15,101 @@ $color-primary-alt-darkest:  darken($color-primary-alt, 70%);
+$font-mono:                  'Roboto Mono', 'Bitstream Vera Sans Mono', 'Consolas', 'Courier', monospace;
+
+// grid
+$grid: 8; // unitless for some calculations
```
What does this value actually stand for? What about the grid?

msecret replied:
Same as the ones below.

thisisdano replied:
this value stands for the base whitespace scale multiplier (i.e. 8px) — it may not have any effect on the columnar "grid"

msecret replied:
Can we call it something like `$base-multiplier` then? Or add a comment next to it to explain that?

thisisdano replied:
ok, I'm changing this to `$grid-base` and commenting the code

msecret commented on src/css/override_vars.scss:
```diff
@@ -14,34 +15,101 @@ $color-primary-alt-darkest:  darken($color-primary-alt, 70%);
+$text-max-width:             36em;
+
+// type scale
+$sans-s1: (8rem / $grid-type);    // 8px;
```
I still think we should move to reversing these values, so low numbers have high values like we were talking about.

thisisdano replied:
I don't really agree (see reasoning above), but if you _must_ insist I will consider.

msecret replied:
Ah, I read your comment too fast, I'm fine with keeping this.

thisisdano replied:
thanks, sorry for the hassle — i don't know why i'm so prickly and pedantic about some issues of naming, but i am!

msecret reviewed (commented):
Also, I'm still confused with the png files you changed. Don't we want those to have transparent backgrounds?

thisisdano commented:
And I don't really care at this point if the PNGs have bgs or not. I can re-export them if it feels important to have them clear.

thisisdano committed:
removed that debug log

thisisdano committed:
varable changes; fixed trans pngs and svgs

thisisdano commented:
OK, fixed those transparent images, too!

thisisdano committed:
tighten up form styles

thisisdano committed:
fixed logo container width

thisisdano committed:
billboard hed padding

thisisdano committed:
tweaking the disclaimer

Commit:
Fix linting errors.

Commit:
Add missing color
